lib/sdotypes.c

```diff
@@ -1246,17 +1246,47 @@ char *sdo_guid_to_string(sdo_byte_array_t *g, char *buf, int buf_sz)
 #endif
 
 /**
- * Write the GID
- * @param sdow - pointer to the struct where the GID is written in JSON format
- * @return none
+ * Write the SigInfo of the form:
+ * SigInfo = [
+ *   sgType: DeviceSgType,
+ *   Info: bstr
+ * ]
+ * @param sdow - pointer to the struct where the GID is to be written.
+ * @return true if write is successfull. false, otherwise.
  */
-void sdo_gid_write(sdow_t *sdow)
+bool sdo_siginfo_write(sdow_t *sdow)
 {
-	sdow_start_array(sdow, 2);
-	sdow_unsigned_int(sdow, SDO_PK_ALGO);
+	bool ret = false;
+	if (!sdow_start_array(sdow, 2)) {
```

Comment:
a comment describing what the 2 means would be helpful OR define a constant for it

Reply:
'2' represents the number of elements in the CBOR array that is specified in the message structure as mentioned in the comments for the method 'sdo_siginfo_write'.
However, individual comments for each method 'sdow_*' and 'sdor_*' will be added at a later time that'd be cleaner.